From dc81ab2729689f570fa04c72496b20ea98792706 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Fri, 18 Mar 2022 23:19:54 -0700
Subject: [PATCH 21/22] monet: Fix overly low chroma for tones below 90

Chroma is capped to 40 for light colors of L* 90 and above for valid
reasons. However, assigning the capped value to "chroma" fails to
account for the fact that chroma is an argument passed to the method,
so the capping effect persists across iterations. This unintentionally
affects and limits the chroma of *all* shades because the first
iteration of the loop is L*=90.

As a result, almost all colors are less chromatic than the intended
chroma >= 48. Fix the issue by assigning capped chroma to a temporary
per-iteration variable.

Change-Id: I18102d0072c89535e27f23fdb98af7c83864ff7e
---
 .../monet/src/com/android/systemui/monet/Shades.java        | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/packages/SystemUI/monet/src/com/android/systemui/monet/Shades.java b/packages/SystemUI/monet/src/com/android/systemui/monet/Shades.java
index aab3538e3c4c..c5b9458614e1 100644
--- a/packages/SystemUI/monet/src/com/android/systemui/monet/Shades.java
+++ b/packages/SystemUI/monet/src/com/android/systemui/monet/Shades.java
@@ -59,10 +59,8 @@ public class Shades {
         shades[1] = ColorUtils.CAMToColor(hue, Math.min(40f, chroma), 95);
         for (int i = 2; i < 12; i++) {
             float lStar = (i == 6) ? MIDDLE_LSTAR : 100 - 10 * (i - 1);
-            if (lStar >= 90) {
-                chroma = Math.min(40f, chroma);
-            }
-            shades[i] = ColorUtils.CAMToColor(hue, chroma, lStar);
+            float shadeChroma = (lStar >= 90) ? Math.min(40f, chroma) : chroma;
+            shades[i] = ColorUtils.CAMToColor(hue, shadeChroma, lStar);
         }
         return shades;
     }
-- 
2.25.1

